components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -1237,9 +1239,9 @@ subroutine Allocation2_ResolveNPLimit (bounds, num_soilc, filter_soilc  , &
                         endif
 
                         plant_nh4demand_vr_patch(p,j) = vmax_plant_nh4(ivt(p))* frootc(p) * froot_prof(p,j) * &
-                             cn_scalar(p) * t_scalar(c,j) * compet_plant_nh4(p)
+                             cn_scalar_runmean(p) * t_scalar(c,j) * compet_plant_nh4(p) 
                         plant_no3demand_vr_patch(p,j) = vmax_plant_no3(ivt(p)) * frootc(p) * froot_prof(p,j) * &
-                             cn_scalar(p) * t_scalar(c,j) * compet_plant_no3(p)
+                             cn_scalar_runmean(p) * t_scalar(c,j) * compet_plant_no3(p)
```

Comment:
@qzhu-lbl @rgknox These fixes are needed in master. Correct?

Reply:
yes